packages/graphql/src/schema/get-where-fields.ts

```diff
@@ -126,17 +126,26 @@ function getWhereFields({
             }
 
             if (["String", "ID"].includes(f.typeMeta.name)) {
-                if (enableRegex) {
+                const matchesSetting: boolean | undefined = features?.filters?.[f.typeMeta.name]?.MATCHES;
+                if (matchesSetting !== undefined) {
+                    if (matchesSetting === true) {
+                        res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
+                    }
+                } else if (enableRegex) {
+                    // Deprecated. To be removed.
```

Comment:
I know you're probably going to do this very soon, but maybe make a proper `TODO` so extensions can find it?

Reply:
Updated it in https://github.com/neo4j/graphql/pull/3076/commits/a5802bb84acb1c02ae25f9abcbe6830942d0b460